File: back-end/app/src/main/java/com/senescyt/app/model/Institucion.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Institucion")
public class Institucion implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 1L;
    /**
     *
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "insId")
    private Long instId;

    @Column(name = "insNombre")
    private String instNombre;
    @Column(name = "intDireccion")
    private String instDireccion;
    @Column(name = "instEstado")
    private int instEstado;
    @Column(name = "instCodigo")
    private String instCodigo;
    @Column(name = "instReferencia")
    private String instReferencia;

    @ManyToOne
    @JoinColumn(name = "tipId", referencedColumnName = "tipId")
    private TipoInstitucion tipId;

    @JsonIgnore
    @OneToMany(mappedBy = "insId")
    private List<Usuario> listaUsuarios;

}

File: front-end/src/app/modelo/Institucion.ts
import { TipoInstitucion } from './tipoInstitucion';

export class Institucion {
  instId: number;
  instNombre: string;
  instDireccion: string;
  instEstado: number;
  instCodigo: string;
  instReferencia: string;
  tipId: TipoInstitucion;

  constructor(
    instId?: number,
    instNombre?: string,
    instDireccion?: string,
    instEstado?: number,
    instCodigo?: string,
    instReferencia?: string,
    tipId?: TipoInstitucion
  ) {
    (this.instId = instId || 0),
      (this.instNombre = instNombre || ''),
      (this.instDireccion = instDireccion || ''),
      (this.instCodigo = instCodigo || ''),
      (this.instReferencia = instReferencia || ''),
      (this.instEstado = instEstado || 0),
      (this.tipId = tipId || new TipoInstitucion());
  }
}
